File: Golang/Problem1.go
package main

type Problem1 struct {
	list []int
	k int
	exp bool
}

func (p1 Problem1) Calculate() bool{
	res := false;
	hashset := map[int]bool{}

	for _, v := range p1.list{
		if hashset[v]{
			res = true
		} else {
			hashset[p1.k-v] = true;
		}
	}
	return res;
}

File: Rust/src/Problem1.rs
pub use problem::problem::ProblemTrait;

pub mod problem1 {

    pub struct Problem1Struct {
        pub list: Vec<i32>,
        pub k: i32,
        pub exp: bool
    }

    use std::collections::HashSet;

    impl super::ProblemTrait for Problem1Struct {
        fn calculate(&self) -> bool{
            let mut res = false;
            let mut set = HashSet::new();
            for x in self.list.iter(){
                if set.contains(x){
                    res = true;
                    return self.exp==res;
                } else {
                    set.insert(self.k-x);
                }

            }
            return self.exp==res;
        }
    }

}

